undertow/src/main/java/org/wildfly/extension/undertow/HostAdd.java

```diff
@@ -69,6 +70,7 @@ protected void performRuntime(OperationContext context, ModelNode operation, Mod
         final boolean isDefaultHost = defaultServerName.equals(serverName) && name.equals(defaultHostName);
         final int defaultResponseCode = HostDefinition.DEFAULT_RESPONSE_CODE.resolveModelAttribute(context, model).asInt();
         final boolean enableConsoleRedirect = !HostDefinition.DISABLE_CONSOLE_REDIRECT.resolveModelAttribute(context, model).asBoolean();
+        DefaultDeploymentMappingProvider.instance().addMapping(defaultWebModule, serverName, name);
```

Comment:
@stuartwdouglas fyi, this is my main question about this. UndertowDeploymentProcessor in the end picks one server name and one host name, so it's not like it's useful to store and use multiple combinations, and picking the last one is as good as anything. But I don't have a deep insight into how this is all meant to work so when you look at this, this aspect is one I wanted to point out.